File: 은지/신규 아이디 추천.py
def solution(new_id):
    answer = ''
    # 1단계
    id = new_id.lower()
    # 변경 아이디 담을 리스트 초기화
    lst_a = []
    # .의 개수 카운트
    cnt = 0
    for x in id:
        # 2단계
        if x == '-' or x == '_' or x == '.' or x.isalpha() or x.isnumeric():
            # 3단계
            if x == '.':
                cnt += 1
            else:
                cnt = 0
            if cnt <= 1:
                lst_a.append(x)
    # 4단계
    if len(lst_a) > 0 and lst_a[0] == '.':
        lst_a.pop(0)
    if len(lst_a) > 0 and lst_a[-1] == '.':
        lst_a.pop(-1)
    # 5단계
    if len(lst_a) == 0:
        lst_a.append('a')
    # 6단계
    if len(lst_a) >= 16:
        lst_a = lst_a[:15]
        if len(lst_a) > 0 and lst_a[-1] == '.':
            lst_a.pop(-1)
    # 7단계
    if len(lst_a) == 1:
        lst_a.append(lst_a[-1])
        lst_a.append(lst_a[-1])
    elif len(lst_a) == 2:
        lst_a.append(lst_a[-1])
    # 리스트 => 문자열 변환
    answer = "".join(lst_a)
    return answer

File: 은지/LV1/신규 아이디 추천.js
function solution(new_id) {
  var answer = '';
  const exam = 'qwertyuiopasdfghjklzxcvbnm0123456789._-'
  var lst_a = []
  var cnt = 0
  // 1단계
  const id = new_id.toLowerCase()
  for (let x of id) {
    // 2단계
    if (exam.indexOf(x) != -1) {
      // 3단계
      if (x === '.') {
        cnt += 1
      } else {
        cnt = 0
      }
      if (cnt <= 1) {
        lst_a.push(x)
      }
    }
  }
  // 4단계
  if (lst_a.length > 0 && lst_a[0] === '.') {
    lst_a.shift()
  }
  if (lst_a.length > 0 && lst_a[lst_a.length-1] === '.') {
    lst_a.pop()
  }
  // 5단계
  if (lst_a.length === 0) {
    lst_a.push('a')
  }
  // 6단계
  if (lst_a.length >= 16) {
    lst_a = lst_a.slice(0, 15)
    if (lst_a[14] == '.') {
      lst_a.pop()
    }
  }
  // 7단계
  if (lst_a.length === 1) {
    lst_a.push(lst_a[0])
    lst_a.push(lst_a[0])
  }
  else if (lst_a.length === 2) {
    lst_a.push(lst_a[1])
  }
  // 배열 => 문자열 변환
  answer = lst_a.join('')
  return answer;
}